Essentials/src/com/earth2me/essentials/commands/Commandtpahere.java

```diff
@@ -32,10 +33,15 @@ public void run(final Server server, final User user, final String commandLabel,
         }
         // Don't let sender request teleport twice to the same player.
         if (user.getConfigUUID().equals(player.getTeleportRequest()) && player.hasOutstandingTeleportRequest() // Check timeout
-            && player.isTpRequestHere() == true) { // Make sure the last teleport request was actually tpahere and not tpa
+            && player.isTpRequestHere()) { // Make sure the last teleport request was actually tpahere and not tpa
             throw new Exception(tl("requestSentAlready", player.getDisplayName()));
         }
         if (!player.isIgnoredPlayer(user)) {
+            TPARequestEvent tpaEvent = new TPARequestEvent(user.getSource(), player, true);
+            ess.getServer().getPluginManager().callEvent(tpaEvent);
+            if (tpaEvent.isCancelled()) {
+                throw new Exception(tl("teleportRequestCancelled", player.getDisplayName()));
```

Comment:
Not sure it should be our job to send a cancel message if another plugin is the one doing the cancelling, and likely has their own reasons for cancelling it.

Reply:
@drtshock I think other plugins can inform players if they want, but we should definitely inform the player ourselves because this is a condition where the teleportation has failed.